src/typed-error.ts

```diff
@@ -1,5 +1,4 @@
-/// <reference path='ext.d.ts'/>
-class BaseError {
+export class BaseError {
```

Comment:
Without the export compilation fails:

```
error TS4020: 'extends' clause of exported class 'TypedError' has or is using private name 'BaseError'.
```

Reply:
FWIW I don't think the `BaseError` class is required since TypeScript 2.2 but I didn't want to "feature creep" the PR too much. Since 2.2 you can use the `new.target` metaproperty referencing the constructor.

I think we'd want something like

```ts
class TypedError extends Error {
    constructor(err?: string | Error) {
        /* ... */
        super(err);
        Object.setPrototypeOf(this, new.target.prototype);
        getStackTrace(this, err);
    }
}
```

[TS 2.2 release notes](https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/release%20notes/TypeScript%202.2.md#support-for-newtarget).

Reply:
Ah, it fails because you're generating a definition file, that kinda sucks :(

As for the 2.2 thing, it does look like it'd do the trick so that'd be cool to use since it's available now - I originally wrote this against 2.1 and never noticed that support was added in 2.2

Reply:
OK, I've gotten rid of the `BaseError` class and extended `Error` directly.